internal/features/stacks/state/stack_meta.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package state
+
+import (
+	"github.com/hashicorp/go-version"
+)
+
+// StackMetadata contains the result of the early decoding of a module,
+// it will be used obtain the correct provider and related module schemas
+type StackMetadata struct {
+	CoreRequirements version.Constraints
```

Comment:
What kind of version constraint is this? Asking because `.terraform-version` is a single specific version and no constraint (I don't know if there's a difference in handling this though)

Reply:
(we removed this here: https://github.com/hashicorp/terraform-ls/pull/1753/commits/b2c17586960f5aaa94e3f863db30701a1907addc)

Reply:
(this can be resolved if/when we merge the proposed edits)